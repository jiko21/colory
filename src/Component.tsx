import React from 'react';

const Component: React.FC = () => {
  const [count, setCount] = React.useState(0);

  const increment = () => {
    setCount(count + 1);
  };

  return (
    <>
      <p>{count}</p>
      <button onClick={increment}>+</button>
    </>
  );
};

export default Component;
