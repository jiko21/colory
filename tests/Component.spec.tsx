import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Component from '@/Component';

describe('Component.tsx', () => {
  it('correctly render', () => {
    const component = render(<Component />);
    expect(component.container).toMatchSnapshot();
  });

  it('correctly fire event', () => {
    render(<Component />);
    fireEvent.click(screen.getByText('+'));
    expect(screen.findByText('1')).toBeTruthy();
  });
});
