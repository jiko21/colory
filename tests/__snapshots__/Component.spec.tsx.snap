// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component.tsx correctly render 1`] = `
<div>
  <p>
    0
  </p>
  <button>
    +
  </button>
</div>
`;
